src/main/java/com/pretz/dyntreeserver/AuthController.java

```diff
@@ -18,23 +19,27 @@ public AuthController(UserService userService) {
         this.userService = userService;
     }
 
-    @RequestMapping(method = GET, path = "/auth")
-    public String login(@RequestParam(value = "name") String name, @RequestParam(value = "password") String password) {
-        User user = new User(name, password);
-        if (userService.checkIfUserExists(user)) {
-            return "User exists!"; //TODO to be changed into some json response for client (?)
-        } else {
-            return "No such user exists!"; //TODO to be changed into some error message to client
+    @RequestMapping(method = POST, path = "/auth")
+    public ResponseEntity<String> login(@RequestBody User user) {
+        boolean userExists = userService.validateUser(user);
+        if (userExists) {
+            //is it RESTful?
+            UserSession userSession = new UserSession(user);
+            return new ResponseEntity<>(HttpStatus.OK);
+        }
+        else {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
```

Comment:
As in `createUser` method - I think it'd be better to throw an exception inside `validateUser` and let it be handled later by an `ExceptionHandler` (as described [here](http://www.baeldung.com/exception-handling-for-rest-with-spring#exceptionhandler) for example).

Reply:
Ok